### ISSUE TITLE
Itch only installs one game from multi-game pack.

### ISSUE THREAD

**Issue by Deozaan** [itchio/itch#563]
The Itch client will only install one game from this [Some Games by Saam](https://saampahlavan.itch.io/some-games-by-saam) package.

This is on Windows 10 using Itch v0.13.2.

But I suppose even if it installed all of them, there'd be no way to select which one to launch...

**Comment by fasterthanlime** [comment]
Picking which executable to launch is coming real quick — however the ability to have several uploads installed in parallel for the same gameID.. isn't planned for now.

The immediate workaround would be to create one page per game. Another solution soon-ish will be to have an upload the page that contains all games & have a manifest in there that lets you pick which game to launch.

(Another solution would be to do like The Basement Collection and have one's own launcher which would allow the dev to give history/context on each game.)

**Comment by Deozaan** [comment]
> however the ability to have several uploads installed in parallel for the same gameID.. isn't planned for now.

How will this affect games with separate 32-bit and 64-bit uploads?

**Comment by fasterthanlime** [comment]
> How will this affect games with separate 32-bit and 64-bit uploads?

The site will get proper architecture tagging & the app will be able to pick the right one — and will allow picking which upload to install in case of ambiguity / if the user asks for it.

There's no point in having _both_ the 32-bit and 64-bit uploads downloaded & installed though.

As a reminder though, the current recommendations (cf. https://docs.itch.ovh/itch/master/integrating/platforms/ ) are:
- 64-bit (or universal) builds for Mac OS/X
- Universal builds for Linux
- 32-bit for Windows

TL;DR: cleaning our architecture act, allowing choice of uploads when it's ambiguous, the point I was making was the having separate uploads installed in _parallel_ is a lot of added complexity and isn't planned for now.

**Comment by Deozaan** [comment]
> There's no point in having both the 32-bit and 64-bit uploads downloaded & installed though.

Right. I figured that. I was just curious how/if the Itch app knew which of the two to install.

> the point I was making was the having separate uploads installed in parallel is a lot of added complexity and isn't planned for now.

Got it. Thanks. :)

**Comment by fasterthanlime** [comment]
> Right. I figured that. I was just curious how/if the Itch app knew which of the two to install.

You're right in that — at this moment — it lacks the data to do that. It could technically use heuristics to distinguish 32-bit uploads from 64-bit uploads but it would be a hack and I wanted to get it done cleanly in the backend instead. (For the same reason that the app will not assume `mac.zip` is an OS/X upload, or `win.zip` is a Windows one. However, wharf channels have auto-tagging from the channel name.. but that's not public for now!).

**Comment by erbridge** [comment]
How are executables chosen currently, if there are multiple ones included?

**Comment by fasterthanlime** [comment]
Choosing the upload: https://github.com/itchio/itch/blob/master/appsrc/tasks/find-upload.js
Choosing the executable: https://github.com/itchio/itch/blob/master/appsrc/tasks/configure.js

**Comment by erbridge** [comment]
Hrmm. So it passes an array of executables (I assume) to the `globalMarket`, but where is the specific one to execute chosen? I'm basically asking so I can reliably work around builds for different architectures.

**Comment by fasterthanlime** [comment]
v25 is out and it lets you install and update separately multiple items from a page.

https://itch.io/app
